File: Java/0977-squares-of-a-sorted-array.java
class Solution {
    public int[] sortedSquares(int[] nums) {
        int length = nums.length;
        int[] res = new int[length];
        int left = 0, right = length - 1;
        for(int i = length - 1; i >= 0; i--){
            if(Math.abs(nums[left]) < Math.abs(nums[right]))
                res[i] = nums[right] * nums[right--];
            else
                res[i] = nums[left] * nums[left++];
        }
        return res;
    }
}

File: Python/0977-squares-of-a-sorted-array.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        res = [0] * len(nums) 
        left = 0
        right  = len(nums) - 1
        for i in reversed(range(len(nums))):
            if abs(nums[left])<abs(nums[right]):
                res[i] = nums[right] ** 2
                right -= 1
            else:
                res[i] = nums[left] ** 2
                left += 1
        return res